[IMP] Manufacturing: Update MPS (#9916)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/workflows/use_mps.rst

   starts. As a result, they manually enter a greater demand in the |MPS|, so they can properly
   replenish the product in time for the increase in customer demand.

.. important::
   It is essential to remember that the |MPS| is a **MANUAL** tool. Adding a product to the |MPS|
   does not cause it to be manufactured or purchased automatically. The |MPS| simply suggests the
   amount of the product that should be replenished, but requires user input to create the |MOs| or
   purchase orders |POs| that are used to replenish it.

   For this reason, it is recommended that the |MPS| **NOT** be used alongside reordering rules for
   the same product. Because reordering rules are an automated workflow, they conflict with the
   manual replenishment method of |MPS|. Using both in unison can lead to inaccurate forecasts and
   the creation of unnecessary replenishment orders.

Enable and configure MPS
========================

To use the |MPS| feature, navigate to :menuselection:`Manufacturing app --> Configuration -->
Settings`, and tick the :guilabel:`Master Production Schedule` checkbox under the

> :guilabel:`Planning` heading. Finally, click :guilabel:`Save`.

Author: You have to save first for the additional fields to appear.

> After enabling the :guilabel:`Master Production Schedule` feature, two new fields appear under it on
> the :guilabel:`Settings` page: :guilabel:`Time Range` and :guilabel:`Number of Columns`.

Reviewer: While what you have is true, and is explained in the subsequent paragraphs, it might be worth being upfront about what these inputs do before going into specifics?
```suggestion
After enabling the :guilabel:`Master Production Schedule` feature, the :guilabel:`Time Range` and :guilabel:`Number of Columns` fields become available for adjusting the time columns on the |MPS|.
```

Reviewer 2: I agree with this suggestion.

place, and offers three options: :guilabel:`Monthly`, :guilabel:`Weekly`, and :guilabel:`Daily`. For
example, if :guilabel:`Monthly` is selected, the |MPS| plans the production requirements of products
and components on a monthly basis.

The :guilabel:`Number of Columns` field is used to specify the quantity of the selected
:guilabel:`Time Range` units shown on the |MPS| page. For example, if the :guilabel:`Time Range`
field is set to :guilabel:`Monthly`, and `12` is entered in the :guilabel:`Number of Columns` field,
the |MPS| shows one column for the next 12 months, starting with the current month.

If the values of the :guilabel:`Time Range` or :guilabel:`Number of Columns` fields are altered,
click :guilabel:`Save` again to save the changes.

.. image:: use_mps/mps-setting.png
   :align: center
   :alt: The MPS setting in the Manufacturing app's settings.

MPS UI
======

To open the |MPS|, navigate to :menuselection:`Manufacturing app --> Planning --> Master Production
Schedule`. The |MPS| view appears as follows:

.. image:: use_mps/mps.png
   :align: center
   :alt: The Master Production Schedule in the Manufacturing app.

> The grey column on the left side of the screen shows a section for every product added to the |MPS|.

Reviewer: While what you have is accurate, I have another perspective to share (:

Instead of just referring to it as a grey column, consider describing it within the context of the entire MPS. For example:

The master production schedule is composed of rows, each containing a product with breakdowns of demand, suggested replenishment, and resulting forecasted stock. The products in the grey column depend on the filters. The columns of the MPS depend on the time range and number of column configurations in the :ref:`settings <line 63>`

The categories shown in the rows of each product's section depend on the filters selected in the
:guilabel:`Search...` bar drop-down menu at the top of the page. The default categories are as
follows:

- :guilabel:`[Product] by [unit]` :icon:`fa-area-chart`: the forecasted stock quantity at the
  beginning of each time period. :guilabel:`[Product]` and :icon:`fa-area-chart` are selectable
  buttons which open the product's page or the forecast report for the product, respectively.
- :guilabel:`- Forecasted Demand`: the demand forecast, which is entered manually. This represents
  an estimate of the demand for the product during each time period.
- :guilabel:`- Indirect Demand Forecast`: while this is a default category, it only appears for
  products that are components of other products. It represents the demand for the component from
  existing MOs.
- :guilabel:`+ Suggested Replenishment`: the quantity of the product that is suggested to be

>   replenished through |MOs| or purchase orders |POs|. To the right of the category title is a

Reviewer: ```suggestion
  replenished through |MOs| or |POs|. To the right of the category title is a
```

>   :guilabel:`Replenish` button, which is used to manually replenish the product, based on the

Reviewer: I totally didn't see that button, so maybe you can add a note or tip admonition block to point it out?
![image](https://github.com/odoo/documentation/assets/23611711/62495bd2-6110-4cd3-b19b-5b48d0ced850)

Reviewer 2: I also think that a quick admonition to help the reader quickly locate this would be super beneficial

  quantity suggested to be replenished.
- :guilabel:`= Forecasted Stock`: the quantity of the product forecasted to be in stock at the end
  of each time period, assuming that suggested replenishment numbers are fulfilled.

Altogether, these default categories form an equation: `Forecasted Demand + Suggested Replenishment
= Forecasted Stock`. The :guilabel:`- Forecasted Demand` and :guilabel:`+ Suggested Replenishment`
fields can be edited for any of the time periods to the right of the product column. Doing so
changes the equation, and updates the value displayed in the :guilabel:`Forecasted Stock` field.

> Changing any of the values also makes an :icon:`fa-times` :guilabel:`reset` button appear to the

Reviewer: I didn't know which value you were talking about at first, so I think it'll be useful to specify here:
```suggestion
In the suggested replenishment line, changing any of the values also makes an :icon:`fa-times` :guilabel:`reset` button appear to the
```

- :guilabel:`Green` means that a replenishment order must be generated to keep stock at the
  :guilabel:`Safety Stock Target`.
- :guilabel:`Gray` means that a replenishment order has already been generated to keep stock at the
  :guilabel:`Safety Stock Target`.
- :guilabel:`Yellow` means that a replenishment order has already been generated, but the quantity
  it was created for is not enough to keep stock at the :guilabel:`Safety Stock Target`.
- :guilabel:`Red` means that a replenishment order has already been generated, but the quantity it
  was created for puts the amount of stock above the :guilabel:`Safety Stock Target`.

The :guilabel:`+ Suggested Replenishment` field appears white if no replenishment order has been
generated, and it is not necessary to generate one at the current moment.

Add a product
=============

To use |MPS| to manage the replenishment of a product, navigate to :menuselection:`Manufacturing app
--> Planning --> Master Production Schedule`. At the top of the |MPS| page, click :guilabel:`Add a
Product` to open the :guilabel:`Add a Product` pop-up window.

.. important::

>    Products must be properly configured to be replenished through |MPS|.

Reviewer: ```suggestion
   Products must be properly configured to be replenished through the |MPS|.
```

   In the case of manufactured products, the :guilabel:`Manufacture` route must be selected in the
   :guilabel:`Routes` section of the :guilabel:`Inventory` tab, on the product's page.

   In the case of products that are purchased, the :guilabel:`Buy` route must be selected in the
   :guilabel:`Routes` section of the :guilabel:`Inventory` tab, on the product's page. Additionally,
   a vendor and the price they sell the product for must also be specified on the

>    :guilabel:`Purchase` tab.

Reviewer: in this section, since you're not explicitly giving instructions to configure the routes of the products, I think italicizing all the items that you've used guilabels for would be the move here. Also, it might be worthwhile to link readers to the relevant documentation pages where specific details on how to configure these are written.

On the pop-up window, select the product to add in the :guilabel:`Product` drop-down menu. If the
product is replenished through manufacturing, select the product's |BoM| in the :guilabel:`Bill of
Materials` field.

.. note::
   Selecting a BoM when adding a product to the |MPS| also adds any components listed on the BoM. If
   it is not necessary to manage the replenishment of components through the |MPS|, simply leave the
   :guilabel:`Bill of Materials` field blank.

> 

Reviewer 2: I know you mentioned that this field didn't pop up for you during your revisions, and I fully saw that -- however, during my 2nd round of revisions, I came across it again. So, I think this might be worth double-checking/looking into... 
![Screenshot 2024-06-26 061658](https://github.com/odoo/documentation/assets/73958186/2e5e9126-df46-4183-bdde-2b394ba3f9da)

In the :guilabel:`Safety Stock Target` field, specify the minimum quantity of the product that
should be kept available for orders at all times. For example, if there should always be 205 units of

> the product available for order fulfillment, enter `20` in the :guilabel:`Safety Stock Target`

Reviewer: "If there should always be 205 units, enter `20` in Safety Stock"? This isn't really making sense to me, is this a typo?

field of the :guilabel:`Add a Product` pop-up window, any components listed on the |BoM| appear on
the page as well.

.. image:: use_mps/add-a-product.png
   :align: center
   :alt: The Add a Product pop-up window in the MPS.

Edit a product
--------------

After adding a product to the |MPS|, it may be necessary to change the replenishment values entered
on the :guilabel:`Add a Product` pop-up window. To do so, click the :guilabel:`# ≤…≤ #` button to
the immediate right of the :guilabel:`Replenish` button, on the :guilabel:`+ Suggested
Replenishment` row, below the product's name.

.. note::
   The first and second number displayed on the :guilabel:`# ≤…≤ #` button correspond to the values
   entered in the :guilabel:`Minimum to Replenish` and :guilabel:`Maximum to Replenish` fields when
   adding the product to the |MPS|.

   For example, if `5` was entered in the :guilabel:`Minimum to Replenish` field, and `100` was
   entered in the :guilabel:`Maximum to Replenish` field, the button appears as :guilabel:`5 ≤…≤
   100`.

   .. image:: use_mps/edit-product-schedule.png
      :align: center
      :alt: The Edit Product Schedule button on the MPS.

> Clicking the :guilabel:`# ≤…≤ #` button opens the :guilabel:`Edit Product Schedule` pop-up window.

Reviewer 2: ```suggestion
Clicking the :guilabel:`# ≤…≤ #` button opens the :guilabel:`Edit Production Schedule` pop-up window.
```

This pop-up window is the same as the :guilabel:`Add a Product` pop-up window, except that the

> :guilabel:`Product` and :guilabel:`Bill of Materials` fields cannot be edited.

Reviewer 2: nor can the 'Production Warehouse' field

On the :guilabel:`Edit Production Schedule` pop-up window, enter the desired values in the

> :guilabel:`Safety Stock Target` :guilabel:`Minimum to Replenish`, and :guilabel:`Maximum to

Reviewer 2: ```suggestion
:guilabel:`Safety Stock Target`, :guilabel:`Minimum to Replenish`, and :guilabel:`Maximum to
```

Replenish` fields. Then, click :guilabel:`Save` to save the changes.

Remove a product
----------------

> To remove a product from the |MPS|, tick the checkbox next to its name. Then, click The

Reviewer 2: ```suggestion
To remove a product from the |MPS|, tick the checkbox to left of its name. Then, click the
```

> :icon:`fa-cog` :guilabel:`Actions` button at the top of the screen, and select the
> :guilabel:`Delete` button.

Reviewer 2: ```suggestion
:icon:`fa-cog` :guilabel:`Actions` button at the top of the screen, and select
:guilabel:`Delete` from the resulting drop-down menu. Then, click :guilabel:`Ok` from the :guilabel:`Confirmation` pop-up window.
```

Deleting a product from the |MPS| removes it, along with all of its data. If the product is

> re-added, its replenishment values must be reconfigured as well.

Reviewer 2: ```suggestion
re-added, its replenishment values must be reconfigured.
```

MPS replenishment
=================

Products in the |MPS| can be replenished in one of three ways:

- Click the :guilabel:`Replenish` button at the top of the screen to generate replenishment orders
  for every product below its :guilabel:`Safety Stock Target` for the current month.
- Click the :guilabel:`Replenish` button on the right side of the :guilabel:`+ Suggested
  Replenishment` row of a specific product, to generate a replenishment order for that specific
  product.
- Tick the checkbox to the left of the product name of one or more products. Then, click the

>   :icon:`fa-cog` :guilabel:`Actions` button at the top of the screen, and select the
>   :guilabel:`Replenish` button. Doing so generates a replenishment order for each selected product.

Reviewer 2: ```suggestion
  :icon:`fa-cog` :guilabel:`Actions` button at the top of the screen, and select
  :guilabel:`Replenish` from the resulting drop-down menu. Doing so generates a replenishment order for each selected product.
```

The type of replenishment order generated corresponds to the route selected on the *Inventory* tab

> of the product's page:

Reviewer 2: ```suggestion
of the product's form:
```

   September, and the store has less than ten Christmas tree |MOs| confirmed for the month of
   December.

   Despite the number of confirmed |MOs|, the procurement manager knows that the demand for
   Christmas trees in December is going to be much higher, once the holiday season starts. As a
   result, they manually enter a greater demand in the |MPS|, so they can properly replenish the
   product in time for the increase in customer demand.

.. important::
   It is essential to remember that the |MPS| is a **MANUAL** tool. Adding a product to the |MPS|
   does not cause it to be manufactured or purchased automatically. The |MPS| simply suggests the
   amount of the product that should be replenished, but requires user input to create the |MOs| or
   |POs| that are used to replenish it.

   For this reason, it is recommended that the |MPS| **NOT** be used alongside reordering rules for
   the same product. Because reordering rules are an automated workflow, they conflict with the
   manual replenishment method of |MPS|. Using both, in unison, can lead to inaccurate forecasts and
   the creation of unnecessary replenishment orders.

> Enable and configure MPS
> ========================

Reviewer 3: Add abbr shorthand to titles :)
```suggestion
Enable and configure |MPS|
==========================
```

After enabling the :guilabel:`Master Production Schedule` feature, two new fields appear under it on
the :guilabel:`Settings` page: :guilabel:`Time Range` and :guilabel:`Number of Columns`.

The :guilabel:`Time Range` field is used to select the period of time over which planning takes
place, and offers three options: :guilabel:`Monthly`, :guilabel:`Weekly`, and :guilabel:`Daily`. For
example, if :guilabel:`Monthly` is selected, the |MPS| plans the production requirements of products
and components on a monthly basis.

The :guilabel:`Number of Columns` field is used to specify the quantity of the selected
:guilabel:`Time Range` units shown on the |MPS| page. For example, if the :guilabel:`Time Range`
field is set to :guilabel:`Monthly`, and `12` is entered in the :guilabel:`Number of Columns` field,
the |MPS| shows one column for the next 12 months, starting with the current month.

If the values of the :guilabel:`Time Range` or :guilabel:`Number of Columns` fields are altered,
click :guilabel:`Save` again to save the changes.

.. image:: use_mps/mps-setting.png
   :align: center
   :alt: The MPS setting in the Manufacturing app's settings.

> MPS UI
> ======

Reviewer 3: Suggestion to use "dashboard" instead of the "UI" user interface acronym that may not be familiar to some readers
```suggestion
|MPS| dashboard
===============
```

  buttons which open the product's page, or the forecast report for the product, respectively.
- :guilabel:`- Forecasted Demand`: the demand forecast, which is entered manually. This represents
  an estimate of the demand for the product during each time period.
- :guilabel:`- Indirect Demand Forecast`: while this is a default category, it **only** appears for
  products that are components of other products. It represents the demand for the component from
  existing MOs.
- :guilabel:`+ Suggested Replenishment`: the quantity of the product that is suggested to be
  replenished through |MOs| or |POs|. To the right of the category title is a
  :guilabel:`Replenish` button, which is used to manually replenish the product, based on the
  quantity suggested to be replenished.

  .. figure:: use_mps/replenish-button.png
     :align: center
     :alt: The Replenish button on the "+ Suggested Replenishment" row.

     The "Replenish" button on the "+ Suggested Replenishment" row.

- :guilabel:`= Forecasted Stock`: the quantity of the product forecasted to be in stock at the end
  of each time period, assuming that suggested replenishment numbers are fulfilled.

> Altogether, these default categories form an equation: `Forecasted Demand + Suggested Replenishment
> = Forecasted Stock`. In the case of components, the :guilabel:`Indirect Demand Forecast` is taken
> into account as well.

Reviewer 3: Suggestion to use the [math directive](https://docutils.sourceforge.io/docs/ref/rst/mathematics.html) here for nice maths-like formatting :)
```suggestion
Altogether, these default categories form an equation:

.. math::
   \text{Forecasted Demand} + \text{Suggested Replenishment} = \text{Forecasted Stock}

In the case of components, the :guilabel:`Indirect Demand Forecast` is taken into account as well.
```

The :guilabel:`- Forecasted Demand` and :guilabel:`+ Suggested Replenishment` fields can be edited
for any of the time periods to the right of the product column. Doing so changes the equation, and
updates the value displayed in the :guilabel:`Forecasted Stock` field.

> Changing the value in the :guilabel:`+ Suggested Replenishment` field also makes an :icon:`fa-times`
> :guilabel:`reset` button appear to the left of the field. Click the :icon:`fa-times` button next to
> any of the fields to reset their value back to the one calculated by the |MPS|.

Reviewer 3: Icon labels :)
```suggestion
Changing the value in the :guilabel:`+ Suggested Replenishment` field also makes an :icon:`fa-times`
:guilabel:`(reset)` button appear to the left of the field. Click the :icon:`fa-times`
:guilabel:`(reset)` button next to any of the fields to reset their value back to the one calculated
by the |MPS|.
```

|MPS| page each time it is opened. If a |BoM| was selected in the :guilabel:`Bill of Materials`
field of the :guilabel:`Add a Product` pop-up window, any components listed on the |BoM| appear on
the page, as well.

.. image:: use_mps/add-a-product.png
   :align: center
   :alt: The Add a Product pop-up window in the MPS.

Edit a product
--------------

After adding a product to the |MPS|, it may be necessary to change the replenishment values entered
on the :guilabel:`Add a Product` pop-up window. To do so, click the :guilabel:`# ≤…≤ #` button to
the immediate right of the :guilabel:`Replenish` button, on the :guilabel:`+ Suggested
Replenishment` row, below the product's name.

.. note::
   The first and second number displayed on the :guilabel:`# ≤…≤ #` button correspond to the values
   entered in the :guilabel:`Minimum to Replenish` and :guilabel:`Maximum to Replenish` fields when
   adding the product to the |MPS|.

   For example, if `5` was entered in the :guilabel:`Minimum to Replenish` field, and `100` was
   entered in the :guilabel:`Maximum to Replenish` field, the button appears as :guilabel:`5 ≤…≤
   100`.

>    .. image:: use_mps/edit-product-schedule.png
>       :align: center
>       :alt: The Edit Product Schedule button on the MPS.

Reviewer 3: IMO your example and notes are sufficient, I think this image is unnecessary 
```suggestion
   .. image:: use_mps/edit-product-schedule.png
      :align: center
      :alt: The Edit Product Schedule button on the MPS.
```

Clicking the :guilabel:`# ≤…≤ #` button opens the :guilabel:`Edit Production Schedule` pop-up
window. This pop-up window is the same as the :guilabel:`Add a Product` pop-up window, except that
the :guilabel:`Product` and :guilabel:`Bill of Materials` fields cannot be edited.

On the :guilabel:`Edit Production Schedule` pop-up window, enter the desired values in the
:guilabel:`Safety Stock Target`, :guilabel:`Minimum to Replenish`, and :guilabel:`Maximum to
Replenish` fields. Then, click :guilabel:`Save` to save the changes.

Remove a product
----------------

To remove a product from the |MPS|, tick the checkbox to the left of its name. Then, click the
:icon:`fa-cog` :guilabel:`Actions` button at the top of the screen, and select :guilabel:`Delete`
from the resulting drop-down menu. Finally, click :guilabel:`Ok` on the :guilabel:`Confirmation`
pop-up window.

Deleting a product from the |MPS| removes it, along with all of its data. If the product is
re-added, its replenishment values must be reconfigured.

> MPS replenishment
> =================

Reviewer 3: ```suggestion
|MPS| replenishment
===================
```
